app/Http/Controllers/ShopController.php

```diff
@@ -2,9 +2,104 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\ShopAddRequest;
+use App\Models\Product;
+use App\Models\User;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
+use App\Models\Shop;
+use Illuminate\Http\Response;
 
 class ShopController extends Controller
 {
-    //
+    public function index()
+    {
+        $shops = Shop::query()->paginate(6);
+        return view('admin.shop.index', [
+            'shops' => $shops,
+        ]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     */
+    public function create()
+    {
+        return view('admin.shop.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(ShopAddRequest $request): RedirectResponse
+    {
+        $shop = Shop::query()->create(
+            $request->all()
+        );
+        return redirect()->route('shop.index')->with(
+            'status',
+            'Shop #' . $shop->id . ' was created!'
+        );
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return Application|Factory|View
+     */
+    public function show(int $id)
+    {
+        return view('admin.shop.show', [
+            "products" => Shop::query()->find($id)->products,
+            "users" => Shop::query()->find($id)->users,
+            'shop' => Shop::query()->find($id)
+        ]);
```

Comment:
Try to reduce requests.
$shop = Shop::query()->find($id);
$users = $shop->users....